### PR TITLE
Whitelisting more builtin primitives

### PR DESCRIPTION
**Pull request by mattrobenolt**
This is also a slight refactor to make it easier to expand on this list
of builtins we allow.

Also, doesn't mutate the global `safe_builtins` dict every time from
`RestrictedPython`, instead, opts for copying the dict first.

### CODE REVIEW

File: redash/query_runner/python.py [hunk 1]
```diff
@@ -38,6 +38,11 @@ def __call__(self):
 
 
 class Python(BaseQueryRunner):
+    safe_builtins = (
+        'sorted', 'reversed', 'min', 'max',
+        'sum', 'set',
```
Comment (mattrobenolt) [inline redash/query_runner/python.py]: To be explicit, `sum` and `set` are the additions here.

**Comment by mattrobenolt** [comment]
It seems these test failures are unrelated since other PRs are failing for the same reason.

**Comment by arikfr** [comment]
Yes, the failures are for a different reason. I'll sort it out. I assume you tested these changes locally?

**Comment by mattrobenolt** [comment]
> I assume you tested these changes locally?

You would be assuming incorrectly. :( I haven't set up re:dash locally to run against. I can do that later. We run it in a docker container ourselves for production.

**Comment by mattrobenolt** [comment]
I'm building docker container locally and testing this out now. In theory, this should work just fine. :)

**Comment by arikfr** [comment]
👍 it's quite easy to run the query runner alone, as it's just Python code.

**Comment by mattrobenolt** [comment]
Alright, couldn't get docker container running, but just got it running through python enough to get into the CLI.

I did:

```
In [1]: from redash.query_runner.python import Python
In [2]: Python({}).run_query('print(sorted, reversed, min, max, sum, set)', None)
Out[2]:
('{"rows": [], "log": ["[2016-11-28T19:38:07.050052] (<built-in function sorted>, <type \'reversed\'>, <built-in function min>, <built-in function max>, <built-in function sum>, <type \'set\'>)"], "columns": []}',
 None)
```

as a test, so it appears that everything is patched in correctly.

fwiw, when building from master in docker, I get the following JavaScript error when trying to go to the `/new/query` page:

image

**Comment by arikfr** [comment]
How do you "build" the frontend code in your Docker image?

Note that I wouldn't recommend using master for "production" usage at the moment. I merged earlier today big refactor of the frontend code, so there are some issues still.

**Comment by mattrobenolt** [comment]
I just did `docker build --rm -t redash .` and ran this which should compile from whatever is in my local directory, so I guess it'd just be this: https://github.com/getredash/redash/blob/master/Dockerfile#L32

**Comment by arikfr** [comment]
Strange, that I would expect to work. I'll give it a look tomorrow.

**Pull request merged**